test/e2e/e2e.go

```diff
@@ -335,6 +335,114 @@ var _ = Describe("validation", func() {
 				})
 			})
 		})
+
+		Context("Correct and inocorrect Metallb resources coexist", func() {
+			var correct_metallb *metallbv1alpha1.Metallb
+			var incorrect_metallb *metallbv1alpha1.Metallb
+			BeforeEach(func() {
+				correct_metallb = &metallbv1alpha1.Metallb{}
+				err := loadMetallbFromFile(correct_metallb, consts.MetallbCRFile)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testclient.Client.Create(context.Background(), correct_metallb)).Should(Succeed())
+
+				incorrect_metallb = &metallbv1alpha1.Metallb{}
+				err = loadMetallbFromFile(incorrect_metallb, consts.MetallbCRFile)
+				Expect(err).ToNot(HaveOccurred())
+				incorrect_metallb.SetName("incorrectname")
+				Expect(testclient.Client.Create(context.Background(), incorrect_metallb)).Should(Succeed())
+			})
+
+			AfterEach(func() {
+				testclient.Client.Delete(context.Background(), incorrect_metallb) // Ignore error, could be already deleted
+				// Check the MetalLB custom resource is deleted to avoid status leak in between tests.
+				Eventually(func() bool {
+					err := testclient.Client.Get(context.Background(), goclient.ObjectKey{Namespace: incorrect_metallb.Namespace, Name: incorrect_metallb.Name}, incorrect_metallb)
+					if errors.IsNotFound(err) {
+						return true
+					}
+					return false
+				}, 1*time.Minute, 5*time.Second).Should(BeTrue(), "Failed to delete MetalLB custom resource")
+
+				err := testclient.Client.Delete(context.Background(), correct_metallb)
+				Expect(err).ToNot(HaveOccurred())
+				// Check the MetalLB custom resource is deleted to avoid status leak in between tests.
+				Eventually(func() bool {
+					err = testclient.Client.Get(context.Background(), goclient.ObjectKey{Namespace: correct_metallb.Namespace, Name: correct_metallb.Name}, correct_metallb)
+					if errors.IsNotFound(err) {
+						return true
+					}
+					return false
+				}, 1*time.Minute, 5*time.Second).Should(BeTrue(), "Failed to delete MetalLB custom resource")
+			})
+			It("should have correct statuses", func() {
+				By("checking MetalLB resource status", func() {
+					Eventually(func() bool {
+						instance := &metallbv1alpha1.Metallb{}
+						err := testclient.Client.Get(context.TODO(), goclient.ObjectKey{Namespace: incorrect_metallb.Namespace, Name: incorrect_metallb.Name}, instance)
+						if err != nil {
+							Expect(err).ToNot(HaveOccurred())
+						}
+						for _, condition := range instance.Status.Conditions {
```

Comment:
Can you wrap this into a function that takes the type and returns the status?

Reply:
done